### DISCUSSION TITLE
Task list checkbox styling

### DISCUSSION THREAD

**Discussion by gudjonss** [ueberdosis/tiptap#3499] [General Questions]
Sorry if this is obvious, but I am having a hard time applying custom styles to the checkbox using the TaskList/TaskItem extension. Just simple things like borders and background color

I'm using @tiptap/react with TailwindCSS 

Could someone point me in the right direction? 🙏

edit: 
I realized I could use tailwindcss [arbitrary variants](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-variants) to style the checkbox element

**Comment by gethari** [comment]
> Sorry if this is obvious, but I am having a hard time applying custom styles to the checkbox using the TaskList/TaskItem extension. Just simple things like borders and background color
> 
> 
> 
> I'm using @tiptap/react with TailwindCSS 
> 
> 
> 
> Could someone point me in the right direction? 🙏
> 
> 
> 
> edit: 
> 
> I realized I could use tailwindcss [arbitrary variants](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-variants) to style the checkbox element

Could you please post your solution with code so that it may help others

**Reply by gudjonss** [reply]
> > Sorry if this is obvious, but I am having a hard time applying custom styles to the checkbox using the TaskList/TaskItem extension. Just simple things like borders and background color
> > I'm using @tiptap/react with TailwindCSS
> > Could someone point me in the right direction? 🙏
> > edit:
> > I realized I could use tailwindcss [arbitrary variants](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-variants) to style the checkbox element
> 
> Could you please post your solution with code so that it may help others

Ofcourse!

At first I used tailwindcss arbitrary variants to style the checkbox, but It quickly become quite the unreadable mess due to having to write the selector for each property, something like this:

``` 
TaskItem.configure({
      HTMLAttributes: {
        class: clsx(
          "flex gap-4",
          "[&_input[type='checkbox']]:some-tailwind-class", 
          "[&_input[type='checkbox']:before]:some-tailwind-class",
          "[&_input[type='checkbox']:checked:before]:some-tailwind-class"",
        ),
      },
    }),
```

Instead of continuing with this I just decided to do this in a separate css file (still using tailwindcss though):

```
.ProseMirror input[type="checkbox"] {
  @apply some-tailwind-classes
}

.ProseMirror input[type="checkbox"]::before {
    @apply some-tailwind-classes
}

.ProseMirror input[type="checkbox"]:checked::before {
    @apply some-tailwind-classes
}
```
